salt/transport/zeromq.py

```diff
@@ -433,7 +433,14 @@ def __del__(self):
     def connect(self):
         if not self.auth.authenticated:
             yield self.auth.authenticate()
-        self.publish_port = self.auth.creds['publish_port']
+
```

Comment:
I'm confused about the relationship between loader thread-safety and these proposed changes to the transports.